src/WebJobs.Script/HostOptionsProvider.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            _extensionOptionsConfigurations = extensionOptionsConfigurations;
+            _serviceProvider = provider;
+            _logger = logger;
+            _concurrencyOptions = concurrencyOption;
+        }
+
+        public string GetPayload()
+        {
+            var payload = new JObject();
+            var extensions = new JObject();
+            var extensionOptions = GetExtensionOptions();
+            foreach (var extension in extensionOptions)
+            {
+                extensions.Add(extension.Key, extension.Value);
+            }
+            if (extensionOptions.Count != 0)
+            {
+                payload.Add("extensions", extensions);
+            }
+
+            var concurrency = GetConcurrencyOption();
+            payload.Add("concurrency", concurrency);
+            return payload.ToString();
+        }
+
+        private Dictionary<string, JObject> GetExtensionOptions()
+        {
+            var configuration = _serviceProvider.GetService<IConfiguration>();
+            // for each of the extension options types identified, create a bound instance
+            // and format to JObject
+            Dictionary<string, JObject> result = new Dictionary<string, JObject>();
+            foreach (IWebJobsExtensionOptionsConfiguration extensionOptionsConfigration in _extensionOptionsConfigurations)
+            {
+                // create the IOptions<T> type
+                Type optionsWrapperType = typeof(IOptions<>).MakeGenericType(extensionOptionsConfigration.OptionType);
+                object optionsWrapper = _serviceProvider.GetService(optionsWrapperType);
+                PropertyInfo valueProperty = optionsWrapperType.GetProperty("Value");
+
+                // create the instance - this will cause configuration binding
+                object options = valueProperty.GetValue(optionsWrapper);
+
+                // get the section name from Extension attribute of IExtensionConfigProvider class
+                string name = extensionOptionsConfigration.ExtensionInfo.ConfigurationSectionName.CamelCaseString();
+
+                IOptionsFormatter optionsFormatter = options as IOptionsFormatter;
+                if (optionsFormatter != null)
+                {
+                    result.Add(name, JObject.Parse(optionsFormatter.Format()).ToCamelCase());
+                }
+                else
+                {
+                    // We don't support the extensions that doesn't implement IOptionsFormatter.
+                    _logger.LogInformation($"{extensionOptionsConfigration.OptionType} doesn't support IOptionsFormatter. Ignored.");
+                }
+            }
+
+            return result;
+        }
+
+        private JObject GetConcurrencyOption()
+        {
+            var concurrencyOption = _concurrencyOptions.CurrentValue;
```

Comment:
So we're going from object to json to JObject then to json again? :) Can't you just return JObject.FromObject(concurrencyOptions), and have the caller do the final camel case on serialization?